sdk/translation/Azure.AI.Translation.Document/tests/GetAllTranslationsFilterTests.cs

```diff
@@ -0,0 +1,164 @@
+            {
+                docsPerJob = 20; // in order to avoid job completing before cancelling
+            }
+            var testDocs = CreateDummyTestDocuments(count: docsPerJob);
+            var sourceContainer = await CreateSourceContainerAsync(testDocs);
+
+            // create a translation job
+            var resultIds = new List<string>();
+            for (int i = 1; i <= jobsCount; i++)
+            {
+                var targetContainer = await CreateTargetContainerAsync();
+                var input = new DocumentTranslationInput(sourceContainer, targetContainer, "fr");
+                var translationOp = await client.StartTranslationAsync(input);
+                resultIds.Add(translationOp.Id);
+                if (jobTerminalStatus == DocumentTranslationStatus.Succeeded)
+                {
+                    await translationOp.WaitForCompletionAsync();
+                }
+                else if (jobTerminalStatus == DocumentTranslationStatus.Cancelled)
+                {
+                    await translationOp.CancelAsync(default);
+                    Thread.Sleep(3000); // wait for cancel status to propagate!
+                }
+            }
+
+            return resultIds;
+        }
+
+        [RecordedTest]
+        public async Task GetAllTranslationStatusesFilterByStatusTest()
+        {
+            // create client
+            var client = GetClient();
+
+            // create test jobs
+            await CreateTranslationJobsAsync(client, jobsCount: 5, docsPerJob: 1, jobTerminalStatus: DocumentTranslationStatus.Succeeded);
+            var cancelledIds = await CreateTranslationJobsAsync(client, jobsCount: 2, jobTerminalStatus: DocumentTranslationStatus.Cancelled);
+
+            // list translations with filter
+            var cancelledStatusList = new List<DocumentTranslationStatus> {
+                    DocumentTranslationStatus.Cancelled,
+                    DocumentTranslationStatus.Cancelling
+            };
+            var filter = new TranslationFilter(
+                statuses: cancelledStatusList
+            );
+            var filteredTranslations = await client.GetAllTranslationStatusesAsync(filter: filter).ToEnumerableAsync();
+            var filteredIds = new HashSet<string>(filteredTranslations.Select(t => t.Id).ToList());
+
+            // assert
+            Assert.That(filteredTranslations.All(t => cancelledStatusList.Contains(t.Status)));
+            Assert.True(cancelledIds.All(cancelledId => filteredIds.Contains(cancelledId)));
+        }
+
+        [RecordedTest]
+        public async Task GetAllTranslationStatusesFilterByIdsTest()
+        {
+            // create client
+            var client = GetClient();
+
+            // create test jobs
+            var allIds = await CreateTranslationJobsAsync(client, jobsCount: 10, docsPerJob: 1, jobTerminalStatus: DocumentTranslationStatus.Succeeded);
+            var targetIds = allIds.GetRange(0, 5);
```

Comment:
again to reduce recording file sizes, it should be the same if you just sent 2 jobs, and then filter by one id